cmd/earthly/registry_cmds.go

```diff
@@ -0,0 +1,273 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/earthly/earthly/cloud"
+	"github.com/moby/buildkit/session/secrets"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+)
+
+func (app *earthlyApp) registryCmds() []*cli.Command {
+	return []*cli.Command{
+		{
+			Name:        "login",
+			Usage:       "Login and store credentials in earthly-cloud",
+			Description: "Login and store credentials in earthly-cloud",
+			UsageText: "earthly registry login --username <username> --password <password> [<host>]\n" +
+				"	earthly registry login --org <org> --project <project> --username <username> --password <password> [<host>]\n",
+			Action: app.actionRegistryLogin,
+			Flags: []cli.Flag{
+				&cli.StringFlag{
+					Name:        "org",
+					EnvVars:     []string{"EARTHLY_ORG"},
+					Usage:       "The organization to which the project belongs.",
+					Required:    false,
+					Destination: &app.orgName,
+				},
+				&cli.StringFlag{
+					Name:        "project",
+					EnvVars:     []string{"EARTHLY_PROJECT"},
+					Usage:       "The organization project in which to store registry credentials, if empty credentials will be stored under the user's secret storage.",
+					Required:    false,
+					Destination: &app.projectName,
+				},
+				&cli.StringFlag{
+					Name:        "username",
+					EnvVars:     []string{"EARTHLY_REGISTRY_USERNAME"},
+					Usage:       "The username to use when logging into the registry.",
+					Required:    true,
+					Destination: &app.registryUsername,
+				},
+				&cli.StringFlag{
+					Name:        "password",
+					EnvVars:     []string{"EARTHLY_REGISTRY_PASSWORD"},
+					Usage:       "The password to use when logging into the registry.",
```

Comment:
Is it worth having a warning here about shell history? I believe that still catches younger developers off-guard. Maybe something like, `CAUTION: if you use this, your password will usually end up in your shell history file in plain text.`

Reply:
I opened #2624 to tackle this, and secrets too.